### ISSUE TITLE
[Enhancement] MultiValidationBehavior

### ISSUE THREAD

**Issue by AndreiMisiukevich** [xamarin/XamarinCommunityToolkit#204]
## Summary
- Combines multiple validation behaviors inside itself.
- IsValid = True if all validators are valid, otherwise IsValid = False
- Probably each validator should contain ErrorMessage property and MultiValidationBehavior should contain the list of current error messages (?) we can make ErrorMessage as attached property

## Intended Use Case
Perform several validations at the same time

**Comment by jfversluis** [comment]
What would this look like in a bit of (pseudo) code? Also maybe in comparison to the validation we have now? You can do the same thing, but simpler, right?

I like the idea. Just wondering what it will look like.

**Comment by AndreiMisiukevich** [comment]
> What would this look like in a bit of (pseudo) code? Also maybe in comparison to the validation we have now? You can do the same thing, but simpler, right?
> 
> I like the idea. Just wondering what it will look like.

```xaml
<Entry>
    <Entry.Behaviors>
        <kit:MultiValidationBehavior IsValid="{Binding IsValid}" ErrorParameters="{Binding ErrorParameters}">
            <kit:FirstValidationBehavior kit:MultiValidationBehavior.ErrorParameter="The First is invalid" />
            <kit:SecondValidationBehavior kit:MultiValidationBehavior.ErrorParameter="The Second is invalid" />
        </kit:MultiValidationBehavior>
    </Entry.Behaviors>
</Entry>
```

@jfversluis smth like this :)

**Comment by sthewissen** [comment]
Maybe something of interest since it combines the errors is to also be able to define a `Separator` for the combination of errors. That way users have a few styling options whether they want them to be separated by commas, new lines, etc.

**Comment by AndreiMisiukevich** [comment]
> Maybe something of interest since it combines the errors is to also be able to define a `Separator` for the combination of errors. That way users have a few styling options whether they want them to be separated by commas, new lines, etc.

Yeah, I thought it would be a separate converter https://github.com/xamarin/XamarinCommunityToolkit/issues/238

**Comment by jfversluis** [comment]
This was merged right, @AndreiMisiukevich ?

**Comment by AndreiMisiukevich** [comment]
Implemented in #239
